core/state.go

```diff
@@ -13,78 +13,80 @@ import (
 
 // Operators
 
-// OperatorSocket is formatted as "host:dispersalPort;retrievalPort;v2DispersalPort"
-type OperatorSocket string
+// OperatorSocket is formatted as "host:dispersalPort;retrievalPort;v2DispersalPort;v2RetrievalPort"
+type OperatorSocket struct {
+	host            string
+	v1DispersalPort string
+	v1RetrievalPort string
+	v2DispersalPort string
+	v2RetrievalPort string
+}
 
-func (s OperatorSocket) String() string {
-	return string(s)
+// NewOperatorSocket returns new OperatorSocket struct
+func NewOperatorSocket(nodeIP, dispersalPort, retrievalPort, v2DispersalPort, v2RetrievalPort string) OperatorSocket {
```

Comment:
nodeIP -> host, it may not be IP but an URL

Reply:
Ack. We internally handle this as a host. ParseOperator also does a net.Lookup of this param to provide better err handling.